dns_sd_avahi.c

```diff
@@ -14,42 +15,115 @@
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
  * Lesser General Public License for more details.
  *
+ * Some of this is insipred from libavahi's example:
+ * https://avahi.org/doxygen/html/client-browse-services_8c-example.html
+ * which is also LGPL 2.1 or later.
+ *
  * */
 
-#include "iio.h"
+#include "iio-private.h"
+#include "network.h"
+#include "iio-lock.h"
 
-#include <errno.h>
-#include <stdbool.h>
-#include <string.h>
 #include <time.h>
 
-#include <avahi-common/error.h>
-#include <avahi-common/simple-watch.h>
-#include <avahi-client/client.h>
-#include <avahi-client/lookup.h>
-
+#include <unistd.h>
 #include "debug.h"
 
-struct avahi_discovery_data {
-	AvahiSimplePoll *poll;
-	AvahiAddress *address;
-	uint16_t *port;
-	bool found, resolved;
-};
+/*
+ * Fundamentally, this builds up a linked list to manage
+ * potential clients on the network
+ */
+
+static int new_discovery_data(struct dns_sd_discovery_data **data)
```

Comment:
this function seems to be defined twice in the same patch;
it makes it a bit hard to follow;

but then also a question: is it meant to be the same thing [in both files] or have different behavior in each?

Reply:
yes, there are some implementation specific (avahi on Linux vs Bonjour on macos) things in the different dns_sd_avahi and dns_sd_bonjour files - some are specific to the library being used, some are specific to the OS. 

The common file dns_sd.c calls them with the same name.

Reply:
ack